scene/gui/tree.cpp

```diff
@@ -3512,7 +3517,33 @@ TreeItem *Tree::_find_item_at_pos(TreeItem *p_item, const Point2 &p_pos, int &r_
 
 			return nullptr;
 		} else {
-			pos.y -= h;
+			if (p_item != root && !p_item->is_collapsed() && !p_item->get_children() && !p_item->get_next() && p_item->get_parent() != root && pos.y < h + cache.vseparation // pos.y < compute_item_height() + 2*vseparation
```

Comment:
```suggestion
			if (p_item != root && !p_item->is_collapsed() && !p_item->get_first_child() && !p_item->get_next() && p_item->get_parent() != root && pos.y < h + cache.vseparation // pos.y < compute_item_height() + 2*vseparation
```
This was renamed.

Also you should remove the commented-out code.